core/executor.go

```diff
@@ -215,12 +219,69 @@ func getExprValueAndType(expr sqlparser.Expr, row DSQLQueryResultRow) (interface
 			return nil, "", fmt.Errorf("unknown column: %s", expr.Name.String())
 		}
 	case *sqlparser.SQLVal:
+		// we currently treat JSON query expression as a string value so we will need to differentiate between JSON and SQL strings
+		if expr.Type == sqlparser.StrVal && strings.HasPrefix(string(expr.Val), "_") && getEncoding(row.Value.TypeEncoding) == OBJ_ENCODING_JSON && getType(row.Value.TypeEncoding) == OBJ_TYPE_JSON {
```

Comment:
You can probably add a comment explaining why we're checking for an underscore prefix here.